synchronization/supervized/supervisor.go

```diff
@@ -0,0 +1,72 @@
+package supervized
+
+import (
+	"context"
+	"fmt"
+	"github.com/orbs-network/orbs-network-go/instrumentation/log"
+	"github.com/pkg/errors"
+	"runtime"
+	"strings"
+)
+
+type Errorer interface {
+	Error(message string, fields ...*log.Field)
+}
+
+func OneOff(logger Errorer, f func()) {
+	go func() {
+		defer recoverPanics(logger)
+		f()
+	}()
+}
+
+func LongLiving(ctx context.Context, logger Errorer, f func()) {
+	defer recoverPanics(logger)
+	go func() {
+		for {
+			f()
+			select {
+			case <-ctx.Done():
+				return
+			default:
+				// repeat
+				//TODO count restarts, fail if too many restarts, etc
+			}
+		}
+	}()
+}
+
+func recoverPanics(logger Errorer) {
+	if err := recover(); err != nil {
+		e := errors.Errorf("goroutine panicked at [%s]: %s", identifyPanic(), err)
```

Comment:
Maybe the second `%s` should be changed to `%v` since we're printing an unknown type